test/remote/gateways/remote_mundipagg_test.rb

```diff
@@ -0,0 +1,155 @@
+    assert_equal 'Simulator|Transação de simulação autorizada com sucesso', response.message
+  end
+
+  def test_successful_purchase_with_more_options
+    options = @options.update({
+      order_id: '1',
+      ip: '127.0.0.1',
+      email: 'joe@example.com',
+      shipping_address: address
+    })
+
+    response = @gateway.purchase(@amount, @credit_card, options)
+    assert_success response
+  end
+
+  def test_successful_purchase_with_voucher
+    @options.update(holder_document: '93095135270')
+    response = @gateway.purchase(@amount, @voucher, @options)
+    assert_success response
+    assert_equal 'Simulator|Transação de simulação autorizada com sucesso', response.message
+  end
+
+  def test_failed_purchase
+    response = @gateway.purchase(105200, @declined_card, @options)
+    assert_failure response
+    assert_equal 'Simulator|Transação de simulada negada por falta de crédito, utilizado para realizar simulação de autorização parcial.', response.message
+  end
+
+  def test_successful_authorize_and_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount, auth.authorization)
+    assert_success capture
+    assert_equal 'Simulator|Transação de simulação capturada com sucesso', capture.message
+  end
+
+  def test_failed_authorize
+    response = @gateway.authorize(105200, @declined_card, @options)
+    assert_failure response
+    assert_equal 'Simulator|Transação de simulada negada por falta de crédito, utilizado para realizar simulação de autorização parcial.', response.message
+  end
+
+  def test_partial_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount-1, auth.authorization)
+    assert_success capture
+  end
+
+  def test_failed_capture
+    response = @gateway.capture(@amount, 'abc')
+    assert_failure response
+    assert_equal 'The requested resource does not exist; Charge not found.', response.message
+  end
+
+  def test_successful_refund
+    purchase = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase
+
+    assert refund = @gateway.refund(@amount, purchase.authorization)
+    assert_success refund
+  end
+
+  def test_partial_refund
+    purchase = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase
+
+    assert refund = @gateway.refund(@amount - 1, purchase.authorization)
+    assert_success refund
+  end
+
+  def test_failed_refund
+    response = @gateway.refund(@amount, 'abc')
+    assert_failure response
+    assert_equal 'The requested resource does not exist; Charge not found.', response.message
+  end
+
+  def test_successful_void
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert void = @gateway.void(auth.authorization)
+    assert_success void
+  end
+
+  def test_successful_void_with_voucher
```

Comment:
👍 Can you add a test for a successful refund with a voucher?

Reply:
Done!